### PR TITLE
feat(FormBuilder): Autosave

### PR DESCRIPTION
**Pull request by SaltyAom**
Add Local Storage for auto-save. Formik will fetch initial value from LocalStorage once loaded.
I'd recommended adding a debounce function.

### CODE REVIEW

File: src/app/components/auth.tsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ const AuthComponent: React.FC<IAuthProps> = props => {
     setIsLoginButtonLoad(true)
 
     const instance = firebase()
-    const provider = new auth.FacebookAuthProvider()
+    const provider = new auth.GoogleAuthProvider()
```
Comment (rayriffy) [inline src/app/components/auth.tsx]: Please change it back to Facebook Provider

**Comment by rayriffy** [comment]
There's still a flaw in this method. How to track the revision of data that being stored in localStorage and data in Firestore.

**Comment by rayriffy** [comment]
also, please make PR checks green.

**Comment by SaltyAom** [comment]
How about adding a timestamp to check which one is the latest data, then overwrite the other one the latest? This would resolve conflict between localStorage and Firestore on multiple devices.

**Comment by rayriffy** [comment]
Yes, that should do it.

**Comment by rayriffy** [comment]
Please issue review request to me when it's ready to merge.

**Comment by SaltyAom** [comment]
@rayriffy, ready merge.

**Pull request merged**